README.md

```diff
@@ -299,6 +299,39 @@ const MyStyledDiv = glamorous.div(
 <MyStyledDiv /> // styles applied: {padding-top: 1, padding-right: 2, padding-bottom: 3, padding-left: 4} and anything coming from `extra-thing`.
 ```
 
+##### withProps
```

Comment:
I still want to think about this before committing, maybe bring in a few people to vote. 

I like both, but they introduce different conventions. Should we be adding the API to _components_, or keep it isolated to the glamorous factory?

There's traits about both approaches that I like. Might pause on this and write up my thoughts comparing both approaches